PolyEngine/Engine/Src/InputWorldComponent.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <Vector.hpp>
+#include "ComponentBase.hpp"
+#include "KeyBindings.hpp"
+
+namespace Poly
+{
+	namespace InputSystem
+	{
+		class InputWorldComponent : public ComponentBase
+		{
+		friend void InputPhase(World*);
+		public:
+			bool GetPressed(eKey key)						 { return CurrKey[static_cast<int>(key)]; }
```

Comment:
All those getters should be const methods. Also consider renaming them from GetPressed to IsPressed, and similarily for other methods.